addons/mail/models/mail_message.py

```diff
@@ -1090,3 +1093,18 @@ def _invalidate_documents(self, model=None, res_id=None):
 
     def _get_search_domain_share(self):
         return ['&', '&', ('is_internal', '=', False), ('subtype_id', '!=', False), ('subtype_id.internal', '=', False)]
+
+    def _is_empty_message(self):
+        self.ensure_one()
+        is_body_empty = not self.body or is_html_empty(html_sanitize(self.body, strip_style=True, strip_classes=True))
+        is_subtype_description_empty = not self.subtype_id or not self.subtype_id.description or is_html_empty(html_sanitize(self.subtype_id.description, strip_style=True, strip_classes=True))
+        return is_body_empty and is_subtype_description_empty and not self.attachment_ids and not self.tracking_value_ids
+
+    def _clean_related_data(self):
+        """ Clean related message data """
+        self.ensure_one()
+        if self.starred_partner_ids:
+            self.write({'starred_partner_ids': [(5, 0, 0)]})
+
+        if self.notification_ids:
+            self.write({'notification_ids': [(5, 0, 0)]})
```

Comment:
I'm not sure but can't we just use `self.starred_partner_ids = False` and `self.notification_ids = False` without anything else?

Reply:
I guess that suggestion might not delete the related records though, so just check if possible otherwise forget it.

Reply:
I found this in the doc and some places in the code and I'm not sure that `self.starred_partner_ids = False` and `self.notification_ids = False` are enough as it may not delete the related records as you mentioned. 
I'll do some tests to check if this is the case.